src/aap_eda/core/utils/rulebook_process_logs.py

```diff
@@ -0,0 +1,73 @@
+#  Copyright 2025 Red Hat, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import re
+from datetime import datetime
+from typing import Optional, Tuple
+
+from django.utils.timezone import make_aware
+
+
+def extract_datetime_and_message_from_log_entry(
+    log_entry: str,
+) -> Tuple[Optional[datetime], str]:
+    """
+    Extract the timestamp and message parts from a log entry.
+
+    Supported timestamp patterns:
+    - "%Y-%m-%d %H:%M:%S,%f"
+    - "%Y-%m-%d %H:%M:%S.%f"
+    - "%Y-%m-%d %H:%M:%S %f"
+    - "** %Y-%m-%d %H:%M:%S.%f"
+
+    Parameters:
+    log_entry (str): The log entry string.
+
+    Returns:
+    Tuple[Optional[datetime], str]: A tuple containing the extracted
+        timestamp as a datetime object (or None if invalid) and the
+        message as a string.
+    """
+    timestamp_patterns = [
+        r"^\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}(?:[,\. ]\d{1,6}) ",
+        r"^\*\* \d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}\.\d{6} ",
+    ]
+    datetime_formats = [
+        "%Y-%m-%d %H:%M:%S,%f",
+        "%Y-%m-%d %H:%M:%S.%f",
+        "%Y-%m-%d %H:%M:%S %f",
+    ]
+
+    for pattern in timestamp_patterns:
+        match = re.match(pattern, log_entry)
```

Comment:
`match` will compile the pattern every time is called and this function will be called tons of times. 
The patterns should be precompiled in the module for a better performance. 
```
TIMESTAMP_PATTERNS = [
    re.compile(r"^\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}(?:[,. ]\d{1,6})?"),
    re.compile(r"^\*\* \d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}\.\d{6}")
]
def extract_datetime_and_message_from_log_entry(
...
```